lib/logstash/outputs/loggly.rb

```diff
@@ -88,31 +110,59 @@ class LogStash::Outputs::Loggly < LogStash::Outputs::Base
 
   public
   def register
-    # nothing to do
+    buffer_initialize(
+      :max_items => @flush_size,
+      :max_interval => @idle_flush_time,
+      :logger => @logger
+    )
   end
 
   public
   def receive(event)
     key = event.sprintf(@key)
     tag = event.sprintf(@tag)
 
+    if expected_field = key[/%{(.*)}/, 1]
```

Comment:
unless loggly throws an error when the key is shaped this way, I'd consider this a breaking change in behaviour as we'll start dropping data that we didn't before.

I see a few options here:
1. leave things as they were before and just log a warning
2. introduce a strict mode (off by default) and flip the switch in a next major version
3. add the option to [DLQ](https://www.elastic.co/guide/en/logstash/current/dead-letter-queues.html) these events (again off by default). This is a lot more work, may not be worth at this point.

Reply:
The API key is actually part of the Loggly URL. So instead of posting to URL `http://logs-01.loggly.com/bulk/API_KEY/tag/tag_name` we would be posting to an invalid URL:

`http://logs-01.loggly.com/bulk/%{key_from_field}/tag/tag_name`. So these events never made it to Loggly, they just caused invalid API calls.

Note that I'm not doing anything to the event, other outputs will receive it. I'm just not enqueuing it to ship to Loggly, to avoid the invalid call.